lib/rubygems/installer.rb

```diff
@@ -698,25 +698,39 @@ def process_options # :nodoc:
 
   def check_that_user_bin_dir_is_in_path # :nodoc:
     user_bin_dir = @bin_dir || Gem.bindir(gem_home)
-    user_bin_dir = user_bin_dir.gsub(File::SEPARATOR, File::ALT_SEPARATOR) if
-      File::ALT_SEPARATOR
 
     path = ENV['PATH']
-    if Gem.win_platform?
+
+    # test folder needs to exist, try a few...
+    test = if    Dir.exist? user_bin_dir  then user_bin_dir
+           elsif Dir.exist? Gem.user_home then Gem.user_home
+           else  __dir__
+           end
+
+    # fs_case_insens is true for case insensitive file systems
+    fs_case_insens =
+      test != test.swapcase && File.identical?(test, test.swapcase)
+
+    # Newer Windows paths can use either ALT_SEPARATOR (\) or SEPARATOR (/)
+    if File::ALT_SEPARATOR
+      include = user_bin_dir.tr(File::ALT_SEPARATOR, File::SEPARATOR)
```

Comment:
(Could the local variable name `include` be something else? Syntax highlighting is confused about its status.)

Reply:
> Syntax highlighting is confused

I hate it when that happens…  I'll update & rebase,  maybe tomorrow...

Reply:
(Also: here's fan-mail about your dedication to keeping Windows + Ruby good: 💌  )

Reply:
@olleolleolle 

Thank you. And thanks for all your work.  Fixed...

Reply:
It's still confused because of `Kernel#test`.